src/lib/chips/chips.scss

```diff
@@ -82,6 +85,7 @@ $mat-chips-chip-margin: 8px;
 .mat-chip-remove {
   margin-right: $mat-chip-remove-margin-after;
   margin-left: $mat-chip-remove-margin-before;
+  cursor: pointer;
```

Comment:
I think that example is mostly to show the color change, rather than the cursor.